src/wazuh_modules/vulnerability_scanner/src/vulnerabilityScannerFacade.cpp

```diff
@@ -81,28 +110,216 @@ void VulnerabilityScannerFacade::start(
         auto scanOrchestrator = std::make_shared<ScanOrchestrator>(
             m_indexerConnector, m_databaseFeedManager, m_reportSocketClient, m_internalMutex);
 
+        m_rebootThread = std::thread(
+            [scanOrchestrator,
+             &stopRetryThread = m_stopRetryThread,
+             &shouldClean = m_shouldClean,
+             &shouldStop = m_shouldStop,
+             &rebootMutex = m_rebootMutex]()
+            {
+                while (!stopRetryThread.load())
```

Comment:
This is a one-time action, an infinite loop until `stopRetryThread` is `true` isn't required.

Reply:
Fix it!